### PR TITLE
Added std:: where it was missing

### PR DESCRIPTION
**Pull request by Crog**
This is in relation to issue #543 and covers the main places where std:: is missing but there may still be further calls.

### CODE REVIEW

**Comment by nabijaczleweli** [comment]
:+1:

**Comment by Crog** [comment]
Okay, so currently Travis doesn't like it if I changed ::tolower to std::tolower. Some other fix is evidently necessary to fix that issue. In my gcc 4.4.2 compiler I am getting:

    /../external/clara.h:235: error: '::tolower' has not been declared

**Comment by kirbyfan64** [comment]
Is there a `#include <ctype.h>`?

**Comment by Crog** [comment]
Good call @kirbyfan64 will add the missing include into clara.h and catch_common.hpp and put the std:: back in :
    #include <cctype>

Is there an easy way to roll back changes on here (I'm fairly new to using GitHub!)

**Comment by nabijaczleweli** [comment]
@Crog You can use the standard set of git commands

**Comment by Crog** [comment]
Thanks, I will take a look at Git commands soon. For now I can't solve the issue of ::tolower as Travis build system doesn't have a test that replicates a strict conforming environment like I have here :(

**Comment by Crog** [comment]
Cracked it. GCC reports nicely in some versions about std::tolower having ambiguous overloads. Fixing the ambiguity with the following tells the compiler which function to use and this now builds under QNX and all the Travis Test  :)

**Comment by nabijaczleweli** [comment]
:+1:

File: include/external/clara.h [hunk 1]
```diff
@@ -232,7 +233,7 @@ namespace Clara {
         }
         inline void convertInto( std::string const& _source, bool& _dest ) {
             std::string sourceLC = _source;
-            std::transform( sourceLC.begin(), sourceLC.end(), sourceLC.begin(), ::tolower );
+            std::transform( sourceLC.begin(), sourceLC.end(), sourceLC.begin(), static_cast<int(*)(int)>(std::tolower) );
```
Comment (Kosta-Github) [inline include/external/clara.h]: this doesn't look good to me...

File: include/internal/catch_common.hpp [hunk 2]
```diff
@@ -22,7 +24,7 @@ namespace Catch {
         return s.find( infix ) != std::string::npos;
     }
     void toLowerInPlace( std::string& s ) {
-        std::transform( s.begin(), s.end(), s.begin(), ::tolower );
+        std::transform( s.begin(), s.end(), s.begin(), static_cast<int(*)(int)>(std::tolower) );
```
Comment (Kosta-Github) [inline include/internal/catch_common.hpp]: same here

Comment (nabijaczleweli) [inline include/internal/catch_common.hpp]: Will the code work with just `&std::tolower` instead of this mostrosity?

Comment (PureAbstract) [inline include/internal/catch_common.hpp]: std::tolower should already be `int std::tolower(int)` - so the cast is a best superfluous, and at worst is undefined behaviour at worst.

Comment (nabijaczleweli) [inline include/internal/catch_common.hpp]: There's also `charT tolower( charT ch, const locale& loc );` in `<locale>`, so we're better off using a lambda anyway

Comment (philsquared) [inline include/internal/catch_common.hpp]: Can't use lambdas due to C++-03 compatibility.
I think the answer might be along similar lines, though - go via a(n out of line) forwarding function.

**Comment by philsquared** [comment]
Apologies for the premature closure - I deleted the branch and it auto-closed all associated PRs - it wasn't intentional.
I also can't re-open it at this point - but I'll come back and review shortly.

File: include/internal/catch_common.hpp (hunk 2 above)
Comment (Crog) [inline include/internal/catch_common.hpp]: From what I can gather this was the only resolution. it is not undefined behavior. It is required to resolve the ambiguity due to the two different impleemntations of std::tolower as @nabijaczleweli  mentions there is a second and is the reason why I had issues. The cast is there to tell the compiler to select the desired function, if there isn't a matching function. 

If it really isn't liked (i wasn't toosure either) then creating a function object which calls std::tolower internally would be a practical way to resolve this.

Comment (PureAbstract) [inline include/internal/catch_common.hpp]: Another reasonable approach might be:
```
int (*tolowep)(int) = std::tolower;
std::transform( s.begin(), s.end(), s.begin(), tolowerp );
```
That will select the correct overload, but avoids actually writing `static_cast`.

**Comment by Crog** [comment]
I have opened this request again under #559

**Pull request closed without merging**